test-case/check-reboot.sh

```diff
@@ -0,0 +1,135 @@
+#!/bin/bash
+
+##
+## Case Name: test reboot
+## Preconditions:
+##    N/A
+## Description:
+##    run reboot for the test
+## Case step:
+##    1. check system status is correct
+##    2. wait for the delay time
+##    3. trigger for the reboot
+## Expect result:
+##    Test execute without report error in the LOG
+##
+
+source $(dirname ${BASH_SOURCE[0]})/../case-lib/lib.sh
+
+OPT_OPT_lst['l']='loop'     OPT_DESC_lst['l']='loop count'
+OPT_PARM_lst['l']=1         OPT_VALUE_lst['l']=3
+
+OPT_OPT_lst['d']='delay'    OPT_DESC_lst['d']='delay after system bootup'
+OPT_PARM_lst['d']=1         OPT_VALUE_lst['d']=2
+
+func_opt_parse_option $*
+
+func_lib_check_sudo
+loop_count=${OPT_VALUE_lst['l']}
+delay=${OPT_VALUE_lst['d']}
+
+# write the orig & status count to the status file
+status_log=$LOG_ROOT/status.txt
+echo $loop_count >> $status_log
+count=$(head -n 1 $status_log|awk '{print $1;}')
+# because write record by desc
+current=$[ $count - $loop_count ]
+
+dlogi "Round: $current/$count"
+
+# verify-pcm-list.sh & verify-tplg-binary.sh need TPLG file
+export TPLG=$(sof-get-default-tplg.sh)
+
+declare -a verify_lst
+verify_lst=(${verify_lst[*]} "verify-firmware-presence.sh")
+verify_lst=(${verify_lst[*]} "verify-kernel-module-load-probe.sh")
+verify_lst=(${verify_lst[*]} "verify-pcm-list.sh")
+verify_lst=(${verify_lst[*]} "verify-sof-firmware-load.sh")
+verify_lst=(${verify_lst[*]} "verify-tplg-binary.sh")
+
+# because for the 1st round those information write by nohup command redirect
+# but after system reboot the nohup redirect is missing
+dlogi "Check status begin"
+for i in ${verify_lst[*]}
+do
+    dlogc "$(dirname ${BASH_SOURCE[0]})/$i"
+    $(dirname ${BASH_SOURCE[0]})/$i
+    if [ $? -ne 0 ];then
+        # last line add failed keyword
+        sed -i '$s/$/ fail/' $status_log
+        dlogi "$i: fail" 
+        exit 1
+    else
+        # last line add pass keyword
+        dlogi "$i: pass" 
+    fi
+done
+dlogi "Round $current: Status check finished"
+sed -i '$s/$/ pass/' $status_log
+[[ $loop_count -le 0 ]] && exit 0
```

Comment:
before exit the whole testing, do you need to restore the original /etc/rc.local that the script modified in later section?

Reply:
the case logic base on `sof-boot-once.sh`  
here use this feature in linux:  
1. `/etc/rc.local` is the as load service  
2. `/etc/rc.local` loaded, modify the file, just effect the file not effect the memory(this round)   

so `sof-boot-once.sh` will add itself into `/etc/rc.local`,  
it means `/etc/rc.local` will load this script without parameter  
`sof-boot-once.sh` load without `/etc/rc.local` will qurey from `/etc/rc.local` for itself  
and delete from `sof-boot-once.sh` to the last line  
So it means boot once, because just effect next boot